File: GFG/Row 105 Given only a pointer/Solution.py
class Node():
    def __init__(self, val = None):
        self.data = val
        self.next = None

def delete_node(node):
    prev = Node()
    if(node == None):
        return
    else:
        while(node.next != None):
            node.data = node.next.data
            prev = node
            node = node.next
        prev.next = None

File: GFG/Row 105 Given only a pointer/Solution.java

public class Solution {

    class Node {
        int data;
        Node next;
        Node(int d)
        {
            data = d;
            next = null;
        }
    }

    public void deleteNode(Node Node_ptr)
    {
        Node temp = Node_ptr.next;
        Node_ptr.data = temp.data;
        Node_ptr.next = temp.next;
        temp = null;
    }
}
